examples/flax/language-modeling/t5_tokenizer_model.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+from typing import Iterator, List, Union
+
+from tokenizers import AddedToken, Regex, Tokenizer, decoders, normalizers, pre_tokenizers, trainers
+from tokenizers.implementations.base_tokenizer import BaseTokenizer
+from tokenizers.models import Unigram
+from tokenizers.processors import TemplateProcessing
+
+
+class SentencePieceUnigramTokenizer(BaseTokenizer):
+    """
+    This class is a copy of `DeDLOC's tokenizer implementation <https://github.com/yandex-research/DeDLOC/blob/main/sahajbert/tokenizer/tokenizer_model.py>`__ .
+
+    Custom SentencePiece Unigram Tokenizer with lower, digits, punctuation and Bengali characters normalization
+    Represents the Unigram algorithm, with the pretokenization used by SentencePiece
+    """
+
+    def __init__(self, replacement: str = "▁", add_prefix_space: bool = True):
```

Comment:
```suggestion
    def __init__(
        self,
        replacement: str = "▁",
        add_prefix_space: bool = True,
        unk_token: Union[str, AddedToken]="<unk>",
        sep_token: Union[str, AddedToken]="</s>",
        pad_token: Union[str, AddedToken]="<pad>",
    ):
        self.special_tokens = {
            "pad": {
                "id": 0,
                "token": pad_token
            },"sep": {
                "id": 1,
                "token": sep_token
            },
            "unk": {
                "id": 2,
                "token": unk_token
            }
        }

        self.special_tokens_list = [None] * len(self.special_tokens)
        for token_dict in self.special_tokens.values():
            self.special_tokens_list[token_dict["id"]] = token_dict["token"]

```

Reply:
Since we need to know the value and identifier of some special tokens for the post-processor and for the training, I suggest to defined them at the tokenizer initialization. With such a proposal, a new user can only change the value of special tokens.

Reply:
Do we have to call the second token `"sep"` token? The model for which I'd like to use the tokenizer for, doesn't make use of `sep_token` as does ALBERT or BERT when passing two sentences. It only accepts a single sentence as input and adds the EOS (end-of-sentence) token to the end. => would it be fine as well to call it `"eos"` instead?

Reply:
So replace all `sep_token` with `eos_token` ?

Reply:
Yes, you are absolutely right! :smile:  It is more correct that it is named `eos` instead of `sep`!